packages/create-llama/templates/types/streaming/nextjs/app/components/chat-section.tsx

```diff
@@ -12,17 +14,22 @@ export default function ChatSection() {
     handleInputChange,
     reload,
     stop,
+    data,
   } = useChat({
     api: process.env.NEXT_PUBLIC_CHAT_API,
     headers: {
       "Content-Type": "application/json", // using JSON because of vercel/ai 2.2.26
     },
   });
 
+  const transformedMessages = useMemo(() => {
+    return transformMessages(messages, data);
```

Comment:
how about this name?
```suggestion
    return insertDataIntoMessages(messages, data);
```